### ISSUE TITLE
addAttachment & addEmbeddedImage ends in site timeout

### ISSUE THREAD

**Issue by LinuxSquare** [PHPMailer/PHPMailer#2114]
Hi there,

I've been working for a while on a website using the `PHPMailer` Library to send Mails.
Now, I wanted to add an embedded Image into the HTML-Message, but the site is always timing out.

The Image I wanted to use has a size of 17.1 KB and the complete website has it's permissions set to `www-data:www-data`.
I'm using `Debian GNU/Linux 9.12 (stretch) x86_64` as my webserver.

The code I tried to use looked the following:
```php
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;
use PHPMailer\PHPMailer\Exception;

try {
   $mail = new PHPMailer(true)
   $mail->SMTPDebug = SMTP::DEBUG_SERVER;
   $mail->isSMTP;
   /* --- Host credentials working */
   $mail->CharSet = 'UTF-8';

   $mail->setFrom('no-reply@myhost.ch', $recipientsName);
   $mail->addAddress($recipientsAddress, $recipientsSurname);

   $mail->isHTML(true);
   $mail->Subject = "Subject";
   $mail->Body = "<img src='cid:coatofarmsberne' alt='Coat of Arms Berne'>";

   /* From now on, it's not working anymore */

   $mail->addAttachment(__DIR__.'/img/coatofarmsberne.jpg');   // I also tried png.
   $mail->addEmbeddedImage(__DIR__.'/img/coatofarmsberne.jpg');   // tried png here as well.

   $mail->send();
} catch(Exception $e) {
   echo "Message could not be sent. Mailer Error: {$mail->ErrorInfo}<br/><br/>";
   echo $e;
}

```

When I comment out the `addAttachment` and `addEmbeddedImage`, it's working and the mail gets sent, but if I leave them uncommented like above, the page just loads and loads for about 10 minutes, until `apache2` says it timed out.

Unfortunately, I can't find any logs, even inside the `/var/log/apache2/site/site.error.log`.

I updated the version from `v6.1.3` to `v6.1.7`, in the hope, it would fix my problem.

I don't know how to proceed any further. It gets even more complicated for me, when I sit in front of my PC and try to find a solution.

I used [this Youtube Video](https://www.youtube.com/watch?v=CRwW3l6dhSE) and the [Wiki](https://github.com/PHPMailer/PHPMailer/wiki/Tutorial#inline-attachments) as a reference.

I thank you all in advance for any help I can get.

Greets
LinuxSquare

**Comment by Synchro** [comment]
That is odd. Not much happens inside `addAttachment` or `addStringAttachment` – all the work goes on at send time. My main suspicion would be on [this line](https://github.com/PHPMailer/PHPMailer/blob/master/src/PHPMailer.php#L2973). Some file systems (particularly Windows network shares) do not play well with functions like `is_file()` and `is_readable()`, or it may be an SELinux issue. I'd recommend setting a debug breakpoint (or just some `echo`s) inside that function and see what it stops on. Does it work in your local dev environment?

**Comment by LinuxSquare** [comment]
Okay, I've added a `die()` inside the method,
```php
public function addAttachment(
        $path,
        $name = '',
        $encoding = self::ENCODING_BASE64,
        $type = '',
        $disposition = 'attachment'
    ) {
        try {
            if (!static::isPermittedPath($path) || !@is_file($path) || !is_readable($path)) {
                //This is only for testing purposes
                die("The Filepath is invalid");
                throw new Exception($this->lang('file_access') . $path, self::STOP_CONTINUE);
            }
```
but still, the site keeps hangig up.
So I guess, if the `die()` didn't execute, the file path should be valid.

**Comment by Synchro** [comment]
Yes, but my point was that it must be one of `isPermittedPath`, `is_file`, or `is_readable` that's causing it. You could test them separately like this:

```php
$a = static::isPermittedPath($path);
var_dump($a, 'a');
$b = is_file($path);
var_dump($b, 'b');
$c = is_readable($path);
var_dump($c, 'c');
try {
            if (!static::isPermittedPath($path) || !@is_file($path) || !is_readable($path)) {
```
So if it doesn't display anything, you'll know it's a, if shows an `a`, you'll know it's b, etc.

**Comment by dcorazolla** [comment]
@LinuxSquare, some solution for this issue?

**Comment by Synchro** [comment]
I suspect this may be fixed by #2069
